### PR TITLE
chore: add cz cli for conventional-commits

### PR DESCRIPTION
**Pull request by wenytang-ms**
add cz for conventional-commits

### CODE REVIEW

**Comment by wenytang-ms** [comment]
this pr enable when developers run git commit -m “……” will trigger an prepare commit msg hook, and cz-cli will give then a conventional-commit rule to select, just like 
image

**Comment by qinezh** [comment]
If I have already used the correct commit message format like `git commit -m "chore: add cz cli for conventional-commits"`, will the hook still be triggered?

File: .husky/exec-git-cz.cmd [hunk 1]
```diff
@@ -0,0 +1,19 @@
+@ECHO off
```
Comment (qinezh) [inline .husky/exec-git-cz.cmd:1]: Looks only *.cmd files checked in, will it work on Linux/macOS?

Comment (wenytang-ms) [reply .husky/exec-git-cz.cmd:1]: default will run by prepare-commit-msg file, in prepare-commit-msg will judge which platform, windows will run cmd, and *nix will run prepare-commit-msg file

**Comment by wenytang-ms** [comment]
> If I have already used the correct commit message format like `git commit -m "chore: add cz cli for conventional-commits"`, will the hook still be triggered?

yes, besides the normal content, cz-cli help to add scope(like "api" or "sdk"), verify import change(to bump up major version), verify issue link.
if we wanna skip this hook with full message, ctr+c will skip this step and go on git commit normally

File: .husky/exec-git-cz.cmd (hunk 1 above)
Comment (qinezh) [reply .husky/exec-git-cz.cmd:1]: got it, thanks for the explanation.

**Comment by qinezh** [comment]
> > If I have already used the correct commit message format like `git commit -m "chore: add cz cli for conventional-commits"`, will the hook still be triggered?
> 
> yes, besides the normal content, cz-cli help to add scope(like "api" or "sdk"), verify import change(to bump up major version), verify issue link.
> if we wanna skip this hook with full message, ctr+c will skip this step and go on git commit normally

The dev experience seems to be not so good for me, if I have already used the right commit message format (including the scope). 

As for the issue link, we could add it in the GitHub PR page and it's not the scope of conventional commits from my understanding:
image

**Comment by wenytang-ms** [comment]
> > > If I have already used the correct commit message format like `git commit -m "chore: add cz cli for conventional-commits"`, will the hook still be triggered?
> > 
> > 
> > yes, besides the normal content, cz-cli help to add scope(like "api" or "sdk"), verify import change(to bump up major version), verify issue link.
> > if we wanna skip this hook with full message, ctr+c will skip this step and go on git commit normally
> 
> The dev experience seems to be not so good for me, if I have already used the right commit message format (including the scope).
> 
> As for the issue link, we could add it in the GitHub PR page and it's not in the scope of conventional commits from my understanding:
> image

ok, it seems a personal choice, as a mandatory project tool is not a better choice, so close this pr.
maybe @LongOddCode  can share this as best practice to team

**Pull request closed without merging**